packages/opentelemetry-tracing/src/utility.ts

```diff
@@ -21,14 +21,31 @@ import {
   DEFAULT_MAX_LINKS_PER_SPAN,
 } from './config';
 import { TracerConfig } from './types';
+import {
+  ParentOrElseSampler,
+  ProbabilitySampler,
+  getEnv,
+  AlwaysOnSampler,
+} from '@opentelemetry/core';
 
 /**
  * Function to merge Default configuration (as specified in './config') with
  * user provided configurations.
  */
 export function mergeConfig(userConfig: TracerConfig) {
   const traceParams = userConfig.traceParams;
-  const target = Object.assign({}, DEFAULT_CONFIG, userConfig);
+  const otelSamplingProbability = getEnv().OTEL_SAMPLING_PROBABILITY;
+  const delegateSampler = new ProbabilitySampler(otelSamplingProbability);
+  const target = Object.assign(
+    DEFAULT_CONFIG,
+    // use default AlwaysOnSampler if otelSamplingProbability is 1
+    otelSamplingProbability !== undefined && otelSamplingProbability < 1
+      ? {
+          sampler: new ParentOrElseSampler(delegateSampler),
+        }
+      : { sampler: DEFAULT_CONFIG.sampler },
```

Comment:
I had to repeat the fallback/default sampler here,  but not sure why (thought that `{}` would be good enough but no strangely).